adafruit_emc2101/__init__.py

```diff
@@ -261,17 +281,23 @@ def external_temperature(self):
         temp_msb = self._ext_temp_msb
         full_tmp = (temp_msb << 8) | temp_lsb
         full_tmp >>= 5
-        full_tmp *= 0.125
-
-        return full_tmp
+        if full_tmp == emc2101_regs.TEMP_FAULT_OPENCIRCUIT or full_tmp == emc2101_regs.TEMP_FAULT_SHORT:
+            if self._sensorfault_exceptions:
+                raise ValueError("External Sensor fault")
+            else:
+                return int(full_tmp)
+        else:
+            full_tmp *= 0.125
+        return float(full_tmp)
 
     @property
     def fan_speed(self):
         """The current speed in Revolutions per Minute (RPM)"""
-
         val = self._tach_read_lsb
         val |= self._tach_read_msb << 8
-        return round(_FAN_RPM_DIVISOR / val, 2)
+        if val < 1:
```

Comment:
I used '< 1' because over many years I have encountered so many bugs relating to such comparisons that I will usually pick the more general comparison (<) to the more specific (=) whenever I can. The actual problem case here is indeed 0, which can happen if something goes wrong with the interface, but I used < 1 just to be sure.  Ditto elsewhere.

Reply:
I see your point, but there really shouldn't be any case where it can return a negative number, based on the code.  It seems overcautious, and I think it would be better for clarity to remove the check.  Same for the others.

Also, are you saying that a result of `0` is actually a problem?

Reply:
I am saying 0 is a problem because we otherwise try to divide by 0.